allennlp/tests/end_to_end/empty_list_test.py

```diff
@@ -0,0 +1,218 @@
+# pylint: disable=no-self-use,invalid-name
```

Comment:
I have pretty mixed feelings about having a separate `end_to_end` module for tests like this - shouldn't this just go in `tests/data/fields/list_field_test.py`?  I understand why you wanted to do it this way, but if the classes you added here had already been in the library for an easy import, you certainly would have just added the test to the `ListField` file.

And as for having an `end_to_end` module in general - basically all of our models have "end to end" tests, and they live in `tests/models/`, so it's not clear what this is for.